src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -164,14 +174,20 @@ std::shared_ptr<Table> SyntheticTableGenerator::generate_table(
             values.push_back(generate_value_by_distribution_type());
           }
 
-          auto value_segment =
-              std::make_shared<ValueSegment<ColumnDataType>>(create_typed_segment_values<ColumnDataType>(values));
+          std::shared_ptr<ValueSegment<ColumnDataType>> value_segment;
+          if (column_specifications[column_index].null_ratio) {
+            value_segment = std::make_shared<ValueSegment<ColumnDataType>>(
+                create_typed_segment_values<ColumnDataType>(values), null_values);
+          } else {
+            value_segment =
+                std::make_shared<ValueSegment<ColumnDataType>>(create_typed_segment_values<ColumnDataType>(values));
+          }
 
           if (!segment_encoding_specs) {
             segments[column_index] = value_segment;
           } else {
-            segments[column_index] = ChunkEncoder::encode_segment(value_segment, column_data_types[column_index],
-                                                                  segment_encoding_specs->at(column_index));
+            segments[column_index] = ChunkEncoder::encode_segment(
+                value_segment, column_specifications[column_index].data_type, segment_encoding_specs->at(column_index));
```

Comment:
Any reason to exclude the encoding in the column specification?

Reply:
No, actually not... Included it in column specification in 710bf3f